### ISSUE TITLE
The API Error prevents theme settings

### ISSUE THREAD

**Issue by baslking** [modufolio/Koken-App#15]
I've done a successful Linux install:
PHP Version 8.1.13
MySQL Version 10.9.4-MariaDB
nginx/1.22.1
ImageMagick 7.1.0
FFmpeg  Version 4.3.4
No matter what I do I always get the error code in api.html  when I try to set a theme.  I've tried all the tricks I can find on Google

I see logs like:
Koken-App/__rewrite_test/index.php" is not found (2: No such file or directory), client: 172.16.16.16, server: _, request: "GET /__rewrite_test/ HTTP/1.1", host: "mynas:8888"
Dec 22 19:39:29 mynas nginx[1201567]: 2022/12/22 18:39:29 [error] 1201567#1201567: *73394 "/poolz/Koken-App/__rewrite_test/index.php" is not found (2: No such file or directory), client: 172.16.16.16, server: _, request: "GET /__rewrite_test/ HTTP/1.1", host: "mynas:8888"
Dec 22 19:39:30  mynas  nginx[1201567]: 2022/12/22 18:39:30 [error] 1201567#1201567: *73321 open() "/poolz/Koken-App/admin/undefined/installs/07a923d2ba871524e7641da96b15b8d2/updates" failed (2: No such file or directory), client: 172.16.16.25, server: _, request: "GET /admin/undefined/installs/07a923d2ba871524e7641da96b15b8d2/updates HTTP/1.1

Otherwise everything seems to work correctly, so close.... Lightroom plugin works etc.

**Comment by modufolio** [comment]
Thank you for reporting this, I'm bit sick now. I will give this priority soon.

**Comment by modufolio** [comment]
Can you test it with the new release 1.1.6?

https://github.com/modufolio/Koken-App/releases/tag/1.1.6

Update instructions

1. Remove all the files except the storage folder.
2. Download the lasted zip https://github.com/modufolio/Koken-App/tags
3. Unpack the zip
4. Copy all the folders / files except the storage folder.
5. Remove all the files in the storage/cache/api folder.


It should work now.

**Comment by modufolio** [comment]
@baslking I see that its working on php 7.4 and not for 8.0 and 8.1.

I will do some more research

**Comment by Jaegerstefan** [comment]
I also have the same issue with the 1.1.6 and PHP8

**Comment by baslking** [comment]
I upgraded to 1.1.7, a new error but Themes are not editable (none appear now) 
I get: 
Console error  Koken isn't able to load data. Please click the button below to try and fix the problem. If you see this error message again thereafter, please [contact us for support](http://help.koken.me/customer/portal/emails/new).

<img width="376" alt="Screenshot 2023-03-11 at 11 13 05" src="https://user-images.githubusercontent.com/404772/224478601-9a9bac61-adaf-40c7-bc13-a837122d1bbf.png">

**Comment by ChemicalShard** [comment]
I’ve had the same error as above but hadn’t acknowledged it. I really appreciate that you are trying to keep koken alive. I’ve tried so many other photography cms but none of them are at the same calibre as koken. It’s a real shame what has happened. On 11 Mar 2023, at 10:13, Brad King ***@***.***> wrote:﻿
I upgraded to 1.1.7, a new error but Themes are not editable (none appear now)
I get:
Console error  Koken isn't able to load data. Please click the button below to try and fix the problem. If you see this error message again thereafter, please contact us for support.

**Comment by modufolio** [comment]
@baslking @Jaegerstefan @ChemicalShard @Accudio Can you test the fix for this issue? In my environment i've solved this issue on PHP 8.1. Added a new version for this fix 1.1.8

**Comment by xabirequejo** [comment]
Not sure if it's the same issue but my hosting provider forced me to upgrade to PHP 8.1 recently. Installed 1.1.8 and I see this

```
API Error
The theme is not able to make contact with your Koken installation. Contact your host to see if they are blocking loopback connections.
```

I was using vanilla Koken until now, maybe I should've update to custom Koken before?

I'm using Dreamhost so I believe that's Apache, MySQL (8.0.28) and PHP 8.1. My skills are limited so not sure how much more details I could provide.

**Comment by baslking** [comment]
Thanks for the rapid updates!  
At first the 1.1.8 didn't change anything, but further investigation shows that just one installed theme was causing the problem.  I removed the "Boulevard" theme and now it seems to be working.  I will keep investigating.  

BTW I am using PHP 8.2 
<img width="1205" alt="Screenshot 2023-03-13 at 00 07 07" src="https://user-images.githubusercontent.com/404772/224579579-40454b1f-cec7-4b5f-9835-d4778465f885.png">

**Comment by modufolio** [comment]
> Not sure if it's the same issue but my hosting provider forced me to upgrade to PHP 8.1 recently. Installed 1.1.8 and I see this
> 
> 
> 
> ```
> 
> API Error
> 
> The theme is not able to make contact with your Koken installation. Contact your host to see if they are blocking loopback connections.
> 
> ```
> 
> 
> 
> I was using vanilla Koken until now, maybe I should've update to custom Koken before?
> 
> 
> 
> I'm using Dreamhost so I believe that's Apache, MySQL (8.0.28) and PHP 8.1.9. My skills are limited so not sure how much more details I could provide.
> 
> 
> 
> Edit: that's when I visit the main site (home). I get the [same message as @baslking](https://github.com/modufolio/Koken-App/issues/15#issuecomment-1464878028) when visiting the admin area.

Do you experience the same issues with a fresh new install too? What themes and plug-ins are you using?

**Comment by ChemicalShard** [comment]
I carried out a fresh install usiong 1.1.8 with the following specs:
- Apache Version | 2.4.56
- PHP Version | 8.1.16
- MySQL Version | 5.7.23-23

I have the same error when I click the 'Site' link within admin panel. I can edit the theme settings but when I click 'Themes' sub-link I get the API error:
<img width="384" alt="Screenshot 2023-03-13 at 12 09 32" src="https://user-images.githubusercontent.com/75095680/224698029-54a259c4-93bf-4195-8162-7a255f4795f5.png">

I tried the following technique @baslking used but the error is still replicated even after clearing system cache:
> Thanks for the rapid updates! At first the 1.1.8 didn't change anything, but further investigation shows that just one installed theme was causing the problem. I removed the "Boulevard" theme and now it seems to be working. I will keep investigating.
> 
> BTW I am using PHP 8.2 <img alt="Screenshot 2023-03-13 at 00 07 07" width="1205" src="https://user-images.githubusercontent.com/404772/224579579-40454b1f-cec7-4b5f-9835-d4778465f885.png">

Other than the above, the whole installation is functioning as expected.

**Comment by Jaegerstefan** [comment]
It's the same for me. The Boulevard themes seems to be problematic, the "original" one and the one i modified in the past.
If only Elementary is available it is mostly stable.

**Comment by xabirequejo** [comment]
Sorry I never replied.

My internet is limited to 0.001 KB/s so it's nearly impossible to test anything like this.

> @xabirequejo Do you experience the same issues with a fresh new install too? What themes and plug-ins are you using?

If by fresh install you mean uploading pictures from scratch, then I didn't go that far. I simply replaced the original Koken with the modified one because the original Koken wasn't compatible with PHP8. But I didn't remove and re-upload my pictures, I left `storage` untouched.

My plugins are:

- font loader
- google analytics (not in use)
- html injector
- password protect

And my themes:

- chastain
- elementary
- ensemble
- repertoire

I tried locating Boulevard but I don't see it. Or am I looking on the wrong place?

I love Koken, I really do, it's perfect for what I want, but I'm not sure it's fixable. Even when it was actively developed it felt slow.

I am now considering moving from a simple hosting to a VPS and in that case I'd try Photoview or Librephoto.
